src/SignalR/common/Shared/ValueTaskExtensions.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Internal
+{
+    internal static class ValueTaskExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Task GetAsTask(this in ValueTask<FlushResult> valueTask)
```

Comment:
Add a comment that this will avoid the allocation of `AsTask` when run in sync because it does not convert to `Task<T>`